src/main.js

```diff
@@ -129,48 +140,241 @@ const drawBackground = () => {
 };
 
 const addMetadata = (_dna, _edition) => {
-  let dateTime = Date.now();
-  let tempMetadata = {
-    name: `${namePrefix} #${_edition}`,
-    description: description,
-    image: `${baseUri}/${_edition}.png`,
-    dna: sha1(_dna),
-    edition: _edition,
-    date: dateTime,
-    ...extraMetadata,
-    attributes: attributesList,
-    compiler: "HashLips Art Engine",
-  };
-  if (network == NETWORK.sol) {
-    tempMetadata = {
-      //Added metadata for solana
-      name: tempMetadata.name,
-      symbol: solanaMetadata.symbol,
-      description: tempMetadata.description,
-      //Added metadata for solana
-      seller_fee_basis_points: solanaMetadata.seller_fee_basis_points,
-      image: `${_edition}.png`,
-      //Added metadata for solana
-      external_url: solanaMetadata.external_url,
-      edition: _edition,
-      ...extraMetadata,
-      attributes: tempMetadata.attributes,
-      properties: {
-        files: [
-          {
-            uri: `${_edition}.png`,
-            type: "image/png",
-          },
-        ],
-        category: "image",
-        creators: solanaMetadata.creators,
-      },
-    };
+  let tempMetadata = {};
+
+  switch (network) {
+    case NETWORK.egld: {
+      tempMetadata = {
+        description: description,
+        dna: sha1(_dna),
+        ...extraMetadata,
+        attributes: attributesList,
+        rarities: {},
+        compiler: "HashLips Art Engine",
+      };
+    }
+    case NETWORK.eth: {
+      tempMetadata = {
+        name: `${namePrefix} #${_edition}`,
+        description: description,
+        image: `${baseUri}/${_edition}.png`,
+        dna: sha1(_dna),
+        edition: _edition,
+        date: Date.now(),
+        attributes: attributesList,
+        ...extraMetadata,
+        compiler: "HashLips Art Engine",
+      };
+      break;
+    }
+    case NETWORK.sol: {
+      tempMetadata = {
+        name: `${namePrefix} #${_edition}`,
+        symbol: solanaMetadata.symbol,
+        description: description,
+        seller_fee_basis_points: solanaMetadata.seller_fee_basis_points,
+        image: `${_edition}.png`,
+        external_url: solanaMetadata.external_url,
+        edition: _edition,
+        ...extraMetadata,
+        attributes: attributesList,
+        properties: {
+          files: [
+            {
+              uri: `${_edition}.png`,
+              type: "image/png",
+            },
+          ],
+          category: "image",
+          creators: solanaMetadata.creators,
+        },
+      };
+      break;
+    }
   }
+
   metadataList.push(tempMetadata);
   attributesList = [];
 };
 
+const addRarityMetadata = () => {
```

Comment:
I think the different implementation can be split into different functions/components, and the don't really belong in `main`, can we refactor a bit?